doozer/doozerlib/cli/release_gen_payload.py

```diff
@@ -604,16 +608,40 @@ def generate_payload_entries(self, assembly_inspector: AssemblyInspector) -> (Di
             # No adjustment for private or public; the assembly's canonical payload content is the same.
 
             entries: Dict[str, PayloadEntry]  # Key of this dict is release payload tag name
-            issues: List[AssemblyIssue]
-            entries, issues = PayloadGenerator.find_payload_entries(assembly_inspector, arch, self.full_component_repo(repo_type=RepositoryType.PUBLIC))
+            payload_issues: List[AssemblyIssue]
+            entries, payload_issues = PayloadGenerator.find_payload_entries(assembly_inspector, arch, self.full_component_repo(repo_type=RepositoryType.PUBLIC))
 
-            # Exclude issues for now, for private repo
-            private_entries, _ = PayloadGenerator.find_payload_entries(assembly_inspector, arch, self.full_component_repo(repo_type=RepositoryType.PRIVATE))
-            entries_for_arch[arch] = entries
+            public_entries: Dict[str, PayloadEntry] = dict()
+            for k, v in entries.items():
+                if not v.build_inspector:
+                    # Its RHCOS, since it doesn't have a build inspector. Put it in for now, but change once RHCOS
+                    # supports private nightlies
+                    public_entries[k] = v
+                    continue
+
+                if v.build_inspector.is_under_embargo() and self.runtime.assembly_type == AssemblyTypes.STREAM:
+                    # It's an embargoed build. Filter it out if its stream
+                    continue
+
+                public_entries[k] = v
+
+            public_entries_for_arch[arch] = public_entries
+            self.assembly_issues.extend(payload_issues)
+
+            # Report issues for any embargoed content being made public.
+            # If releasing after embargo lift, these can be permitted using 'EMBARGOED_CONTENT' code
+            embargo_issues = PayloadGenerator.embargo_issues_for_payload(public_entries, arch)
+            self.assembly_issues.extend(embargo_issues)
+
+            private_entries, private_payload_issues = PayloadGenerator.find_payload_entries(assembly_inspector, arch, self.full_component_repo(repo_type=RepositoryType.PRIVATE))
             private_entries_for_arch[arch] = private_entries
-            self.assembly_issues.extend(issues)
 
-        return entries_for_arch, private_entries_for_arch
+            # Check to see if there are private only issues, and add them to the list of assembly issues
+            private_only_issues = set(private_payload_issues) - set(payload_issues)
+
+            self.assembly_issues.extend(list(private_only_issues))
```

Comment:
i debated whether it made sense to add these at all. seems like it could be confusing if there were any and they showed up on the public payloads. OTOH they would show issues that could become public if the payload were promoted so... i guess it makes sense to include them.